src/DotNet.Extensions.DependencyInjection/ServiceChangeToken.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
+
+namespace DotNet.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Implements <see cref="IChangeToken"/>
+    /// </summary>
+    public class ServiceChangeToken : IChangeToken, IDisposable
+    {
+        private CancellationTokenSource _cts = new CancellationTokenSource();
```

Comment:
Consider a default timeout?

Reply:
The cancellation of the token itself is what is used to propagate the IChangeToken notifications. So it wouldn't be good to have that happen automatically.